Modix.Services/Core/AuthorizationService.cs

```diff
@@ -191,26 +191,44 @@ public async Task RemoveClaimMapping(IGuildUser user, ClaimMappingType type, Aut
         }
 
         /// <inheritdoc />
-        public async Task<IReadOnlyCollection<AuthorizationClaim>> GetGuildUserClaimsAsync(IGuildUser guildUser)
+        public async Task<IReadOnlyCollection<AuthorizationClaim>> GetGuildUserClaimsAsync(IGuildUser user)
         {
-            if (guildUser == null)
-                throw new ArgumentNullException(nameof(guildUser));
+            if (user == null)
+                throw new ArgumentNullException(nameof(user));
 
-            if (guildUser.Id == DiscordClient.CurrentUser.Id)
+            if (user.GuildPermissions.Administrator)
                 return Enum.GetValues(typeof(AuthorizationClaim)).Cast<AuthorizationClaim>().ToArray();
 
-            if (guildUser.Id == CurrentUserId)
-                return CurrentClaims;
+            var claims = new HashSet<AuthorizationClaim>();
 
-            return await GetGuildUserCurrentClaimsAsync(guildUser.GuildId, guildUser.RoleIds, guildUser.Id);
+            foreach (var claimMapping in (await ClaimMappingRepository
+                .SearchBriefsAsync(new ClaimMappingSearchCriteria()
+                {
+                    GuildId = user.Guild.Id,
+                    RoleIds = user.RoleIds,
+                    UserId = user.Id,
+                    IsDeleted = false
+                }))
+                // Evaluate role mappings (userId is null) first, to give user mappings precedence.
+                .OrderBy(x => x.UserId)
+                // Evaluate granted mappings first, to give denied mappings precedence.
+                .ThenBy(x => x.Type))
+            {
+                if (claimMapping.Type == ClaimMappingType.Granted)
+                    claims.Add(claimMapping.Claim);
+                else
+                    claims.Remove(claimMapping.Claim);
+            }
+
+            return claims;
         }
 
         /// <inheritdoc />
-        public async Task OnAuthenticatedAsync(ulong guildId, IEnumerable<ulong> roleIds, ulong userId)
+        public async Task OnAuthenticatedAsync(IGuildUser user)
```

Comment:
So, what ended up happening here was that I was having an issue where trying to access `CurrentClaims` for a user was returning the memoized result of the internal call straight from the database - and not taking into account the override for admins/self. So, I decided it would be best just to merge the two calls into one (as at the time, it was only used in one place). But, if you're working on it still, I'll absolutely revert that.

As for taking an IGuildUser, in the webserver, we actually only store the user id in the login token. For authenticated requests (on controllers deriving from ModixController), we retrieve a (usually cached) user from the DiscordSocketClient and pass that to service calls, etc - so I figured I would reduce the parameters since we would always have an IGuildUser instance.